### PR TITLE
feat: add the option to disable workflow

### PR DESCRIPTION
**Pull request by nejcr**
Closes https://github.com/keephq/keep/issues/1739

## 📑 Description

Added `is_disabled` to the `workflow` table and introduced `disabled` in the YAML configuration. Updated documentation and UI; see screenshots 

## ✅ Checks
- [ x ] My pull request adheres to the code style of this project
- [ x ] My code requires changes to the documentation
- [ x ] I have updated the documentation as required
- [ x ] All the tests have passed

## ℹ Additional Information
image
image

Also spent an hour or two figuring out how to run `keep` in PyCharm; I might open a separate PR to document the process."

 /claim https://github.com/keephq/keep/issues/1739

### CODE REVIEW

**Comment by talboren** [comment]
Not sure if I'm right but it might conflict with: https://github.com/keephq/keep/pull/1488 which is about to get merged.

**Comment by nejcr** [comment]
> Not sure if I'm right but it might conflict with: #1488 which is about to get merged.

I can wait, looking at it, it would be easier for me if I resolved conflicts afterwards

**Comment by rajeshj11** [comment]
@nejcr I think your changes are conflicting with my changes. please resolve them. post-resolution can you also share the final changes with me? If needed any clarification on new changes please do reach out to me.

**Comment by nejcr** [comment]
@shahargl I have resolved the conflicts with the main branch.

@rajeshj11 I incorporated some of your validator changes as they were necessary for proper functionality. The code should now be free of conflicts. Additionally, I simplified the nested ternary logic for better readability.

File: keep-ui/app/workflows/builder/builder-validators.tsx [hunk 1]
```diff
@@ -17,10 +17,11 @@ export function globalValidatorV2(
...
   if (
-    !!definition?.properties && 
-    !('manual' in definition.properties) && 
-    !('interval' in definition.properties) && 
-    !('alert' in definition.properties)
+    !!definition?.properties &&
+    !definition.properties['manual'] &&
+    !definition.properties['interval'] &&
+    !definition.properties['alert'] &&
+    !definition.properties['disabled']
```
Comment (rajeshj11) [inline keep-ui/app/workflows/builder/builder-validators.tsx:24]: I think this check is not needed. this is not trigger related data.
!definition.properties['disabled']

**Comment by nejcr** [comment]
Thank you, @Matvey-Kuk.

I have rebased the changes from the main branch to keep this PR up to date.

Please let me know if anything else is needed. (Note: There were some database setup issues with the tests running on GitHub Actions.)

**Comment by Matvey-Kuk** [comment]
@nejcr I see the tests are still failing, please let me know if you are planning to fix them

**Comment by Matvey-Kuk** [comment]
Great work, thank you!

**Comment by nejcr** [comment]
@Matvey-Kuk @talboren another migration got into main before aproved workflows were triggered.
I've had to resolve another double headed migration.

**Comment by nejcr** [comment]
@Matvey-Kuk looks like something changed so the now the pr tests are red.
I have added a possible fix(passes on 1739 branch).

**Review by Matvey-Kuk — APPROVED** [review]

**Comment by Matvey-Kuk** [comment]
Thank you!

**Pull request merged**